### ISSUE TITLE
Touchpad scrolling not working on Wayland

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura#92]
*On GitLab by @ReekyMarko on Oct 18, 2018, 18:29*

---

I'm running Gnome on Wayland and touchpad scrolling doesn't seem to be working. Tested on Gnome with Xorg and it works fine. I can provide more information, I just don't know what is needed.

**Comment by sebastinas** [comment]
*On GitLab by @kflak on Nov 20, 2018, 14:47*

---

I can confirm this. Running Sway/Arch Linux rt-lts on a macbook pro 2013, had no problem before I made the switch to wayland.

**Comment by sebastinas** [comment]
*On GitLab by @exastion on Nov 23, 2018, 10:54*

---

The same issue for me.
```
$ zathura --version
zathura 0.4.1
girara 0.3.1 (runtime: 0.3.1)
(plugin) pdf-mupdf (0.3.4) (/usr/lib/zathura/libpdf-mupdf.so)
```
```
OS: Arch Linux 
Kernel: x86_64 Linux 4.19.2-arch1-1-ARCH
Uptime: 3d 16h 44m
Packages: 671
Shell: zsh 5.6.2
Resolution: 2560x1440
WM: sway
CPU: Intel Core i5-6200U @ 4x 2.8GHz [31.0°C]
GPU: HD Graphics 520
RAM: 6114MiB / 7859MiB
```
Running on a Lenovo T460s
Using a mouse everything works fine, other applications work fine too.

**Comment by sebastinas** [comment]
I'm not running any Wayland installations, so I can't test it. However, we simply rely on Gtk sending the right events. If Gtk doesn't send them, I don't think we can do much in zathura to fix that.

**Comment by sebastinas** [comment]
*On GitLab by @exastion on Nov 23, 2018, 11:29*

---

I did some quick research, apparently, gtk only fires GDK_SCROLL_SMOOTH for Touchpad scroll and not GDK_SCROLL.
If I get the time I will try to look into it.

**Comment by sebastinas** [comment]
Thanks, that'd be great.

**Comment by sebastinas** [comment]
*On GitLab by @kflak on Nov 26, 2018, 11:24*

---

Much appreciated!

**Comment by sebastinas** [comment]
*On GitLab by @mdeff on Feb 1, 2019, 18:32*

---

Same issue here (with sway on Arch Linux).

For people looking for a fix while the problem is being fixed, add `set smooth-scroll true` to `.config/zathura/zathurarc`. (Fixes scroll and zoom.)

@exastion has the bug been reported to GDK?

**Comment by sebastinas** [comment]
*On GitLab by @kflak on Feb 3, 2019, 14:38*

---

Thanks @mdeff for the workaround! Works nicely for me (also on arch/sway)

**Comment by sebastinas** [comment]
*On GitLab by @exastion on Feb 3, 2019, 14:45*

---

I tried this in zathura directly, there it doesn't work. Setting it in the `zathurarc` works, thanks.

I had not much time to investigate further...

I did not report anything to GDK as for me this behavior is quite reasonable.

In my opinion, the only issue is then, that `set smooth-scroll true` does not work from within zathura. Perhaps it should also be stated somewhere, that Touchpad scroll is smooth.
